dot/services_test.go

```diff
@@ -100,54 +214,124 @@ func Test_createCoreService(t *testing.T) {
 		dh  *digest.Handler
 	}
 	tests := []struct {
-		name    string
-		args    args
-		want    *core.Service
-		wantErr bool
+		name string
+		args args
+		want *core.Service
+		err  error
 	}{
-		// TODO: Add test cases.
+		{
+			name: "missing keystore test",
+			args: args{
+				cfg: cfg,
+				st:  stateSrvc,
+			},
+			err: errors.New("cannot have nil keystore"),
+		},
+		{
+			name: "working example",
+			args: args{
+				cfg: cfg,
+				ks:  ks,
+				st:  stateSrvc,
+			},
+			want: &core.Service{},
+		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			got, err := createCoreService(tt.args.cfg, tt.args.ks, tt.args.st, tt.args.net, tt.args.dh)
-			if (err != nil) != tt.wantErr {
-				t.Errorf("createCoreService() error = %v, wantErr %v", err, tt.wantErr)
-				return
+			if tt.err != nil {
+				assert.EqualError(t, err, tt.err.Error())
+			} else {
+				assert.NoError(t, err)
 			}
-			if !reflect.DeepEqual(got, tt.want) {
-				t.Errorf("createCoreService() got = %v, want %v", got, tt.want)
+			if tt.want != nil {
+				assert.NotNil(t, got)
 			}
 		})
 	}
 }
 
 func Test_createDigestHandler(t *testing.T) {
+	cfg := NewTestConfig(t)
+	require.NotNil(t, cfg)
```

Comment:
nit I don't think this NotNil is really required since (afaik) we are asserting something from the test (`NewTestConfig`). Same for below checks as well. Although if these helper functions do use some production code, maybe do the assertion inside them since you are already passing `t` to them.

Reply:
Ok, removed.